### DISCUSSION TITLE
SLEIGH: How to detect if instruction is in delay slot?

### DISCUSSION THREAD

**Discussion by pjsoberoi** [NationalSecurityAgency/ghidra#5905] [Q&A]
I'm looking at an instruction that behaves differently if it is executing within a delay slot. How would I detect if an instruction is running within a delay slot? Thanks.

**Comment by GhidorahRex** [comment]
You might have to set a context bit for this. Does this processor allow jumps to instructions in delay slots?

**Reply by pjsoberoi** [reply]
Sorry lost track of this thread. The architecture is SuperH2:

![image](https://github.com/NationalSecurityAgency/ghidra/assets/60081609/82c7cc30-c03d-4872-b439-a43895719a95)

I'll check if it the processor allows jumps to instructions in delay slots but I assume that to be true.
